File: client-scalaz/src/main/scala/org/openapitools/client/api/Step.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.OffsetDateTime

import Step._

case class Step (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
owner: Option[Account],
deviceId: Option[String],
leg: Option[Leg],
distance: Option[Double],
duration: Option[Long],
startLatitude: Option[Double],
startLongitude: Option[Double],
startAltitude: Option[Double],
startDate: Option[OffsetDateTime],
endLatitude: Option[Double],
endLongitude: Option[Double],
endAltitude: Option[Double],
endDate: Option[OffsetDateTime])

object Step {
  import DateTimeCodecs._

  implicit val StepCodecJson: CodecJson[Step] = CodecJson.derive[Step]
  implicit val StepDecoder: EntityDecoder[Step] = jsonOf[Step]
  implicit val StepEncoder: EntityEncoder[Step] = jsonEncoderOf[Step]
}

File: client-clojure/src/sirqul_io_t_platform/specs/step.clj
(ns sirqul-io-t-platform.specs.step
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.account :refer :all]
            [sirqul-io-t-platform.specs.leg :refer :all]
            )
  (:import (java.io File)))


(def step-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :owner) account-spec
   (ds/opt :deviceId) string?
   (ds/opt :leg) leg-spec
   (ds/opt :distance) float?
   (ds/opt :duration) int?
   (ds/opt :startLatitude) float?
   (ds/opt :startLongitude) float?
   (ds/opt :startAltitude) float?
   (ds/opt :startDate) inst?
   (ds/opt :endLatitude) float?
   (ds/opt :endLongitude) float?
   (ds/opt :endAltitude) float?
   (ds/opt :endDate) inst?
   })

(def step-spec
  (ds/spec
    {:name ::step
     :spec step-data}))
